public/app/features/alerting/unified/api/buildInfo.ts

```diff
@@ -61,13 +62,22 @@ export async function discoverDataSourceFeatures(dsSettings: {
   // The current implementation of Loki's build info endpoint is useless
   // because it doesn't provide information about Loki's available features (e.g. Ruler API)
   // It's better to skip fetching it for Loki and go the Cortex path (manual discovery)
-  const buildInfoResponse = type === 'loki' ? undefined : await fetchPromBuildInfo(url);
 
+  let fallBackInfoResponse: PromBuildInfoSeriesResponse | undefined;
+  const isLoki = type === 'loki';
+  const buildInfoResponse = isLoki
```

Comment:
This is all fragile and confusing, `fetchPromBuildInfo()` returns undefined (and no error) if it gets a 404, which means that the application is Cortex (called Lotex in the code), but if it throws an error with a non 404 code (like the 405s I'm getting from prometheus < 2.14), then we want to try to call the query api to get the series build info. Ugh. 
Is there a way we can clean this up and maybe not use a functions return of undefined as an indication that the function returned valid output? That might be a good place to start.

Reply:
This is also used in discoverAlertmanagerFeaturesByUrl, and refactor is not as easy as I'd hoped. I'll try to come back to this later.